integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -217,41 +217,45 @@ private int getattrInternal(String path, FileStat stat) {
       }
 
       Optional<URIStatus> status = AlluxioFuseUtils.getPathStatus(mFileSystem, uri);
-      if (!status.isPresent()) {
-        LOG.debug("Failed to getattr {}: path does not exist or is invalid", path);
-        return -ErrorCodes.ENOENT();
-      }
-      AlluxioFuseUtils.fillStat(mAuthPolicy, stat, status.get());
-
-      if (!status.get().isCompleted()) {
-        Set<FuseFileEntry<FuseFileStream>> fuseStreams
-            = mFileEntries.getByField(PATH_INDEX, path);
-        boolean hasWriteStream = false;
-        if (!fuseStreams.isEmpty()) {
-          for (FuseFileEntry<FuseFileStream> stream : fuseStreams) {
-            FileStatus fileStatus = stream.getFileStream().getFileStatus();
-            if (fileStatus instanceof  CreateFileStatus) {
-              // should have only one
-              AlluxioFuseUtils.updateCreateFileStatus(stat, (CreateFileStatus) fileStatus);
-              hasWriteStream = true;
+      status.ifPresent(uriStatus -> AlluxioFuseUtils.fillStat(mAuthPolicy, stat, uriStatus));
+
+      boolean hasWriteStream = false;
+      Set<FuseFileEntry<FuseFileStream>> fuseStreams
+          = mFileEntries.getByField(PATH_INDEX, path);
+      if (!fuseStreams.isEmpty()) {
+        for (FuseFileEntry<FuseFileStream> stream : fuseStreams) {
+          FileStatus fileStatus = stream.getFileStream().getFileStatus();
+          if (fileStatus instanceof CreateFileStatus) {
+            // File is being written by current Alluxio client, should be only one write stream
+            if (status.isPresent()) {
+              AlluxioFuseUtils.updateStatSize(stat, fileStatus.getFileLength());
+            } else {
+              AlluxioFuseUtils.fillStat(stat, (CreateFileStatus) fileStatus);
             }
+            hasWriteStream = true;
           }
         }
-        if (!hasWriteStream) {
-          // File is being written by other Alluxio client
-          status = AlluxioFuseUtils.waitForFileCompleted(mFileSystem, uri);
-          status.ifPresent(uriStatus
-              -> AlluxioFuseUtils.updateStatSize(stat, uriStatus.getLength()));
-          if (!status.isPresent()) {
-            LOG.error("File {} is not completed, cannot get accurate file length", path);
-          }
+      }
+
+      if (!hasWriteStream && status.isPresent() && !status.get().isCompleted()) {
+        // File is being written by other Alluxio client
+        Optional<URIStatus> completedFileStatus = AlluxioFuseUtils
+            .waitForFileCompleted(mFileSystem, uri);
+        completedFileStatus.ifPresent(uriStatus
+            -> AlluxioFuseUtils.updateStatSize(stat, uriStatus.getLength()));
+        if (!completedFileStatus.isPresent()) {
+          LOG.error("File {} is not completed, cannot get accurate file length", path);
         }
       }
+
+      if (!status.isPresent() && !hasWriteStream) {
```

Comment:
Fix the difference between Alluxio and UFS that
UFS files are invisible until finished written (cannot get file status).
Need to check current write streams to understand whether the UFS file is being written by the current FUSE.

Reply:
what if the user creates a same file before completion?

Reply:
Sorry that miss this question.
Multiple scenarios may happen:
1. User wants to create the same file through the same FUSE. the user needs to get a write lock to be able to write to the same file and will wait for the previous one to finish.
2. User creates the same file outside this FUSE, through other Alluxio Fuse or other methods. In this case, i assume only one write will success. If the current FUSE, then no problem, if other ways, the current fuse write will be errored out and FUSE kernel will cause release to close the write stream and unlock.